src/python/bot/fuzzers/afl/launcher.py

```diff
@@ -256,29 +254,60 @@ class FuzzingStrategies(object):
   FAST_CAL_RANDOM_STRATEGY = 'strategy_fast_cal_random'
   FAST_CAL_MANUAL_STRATEGY = 'strategy_fast_cal_manual'
 
-  def __init__(self, target_path):
-    strategy_pool = strategy_selection.generate_weighted_strategy_pool(
-        strategy_list=strategy.AFL_STRATEGY_LIST,
-        use_generator=True,
-        engine_name='afl')
-    # Select a generator to attempt to use for existing testcase mutations.
-    self.candidate_generator = engine_common.select_generator(
-        strategy_pool, target_path)
+  # TODO(mbarbella): The codepath involving |strategy_dict| and the
+  # to_strategy_dict function should be removed when everything is fully
+  # converted to the engine pipeline. For now this allows the code to be shared
+  # between both cases while still adhering to the new pipeline's API.
+  def __init__(self, target_path, strategy_dict=None):
+    self.corpus_subset_size = None
+
+    # If we have already generated a strategy dict, use that in favor of
+    # creating a new pool and picking randomly.
+    if strategy_dict:
+      self.candidate_generator = strategy_dict['candidate_generator']
```

Comment:
should we use .get() here, or are these always expected to exist?

Reply:
This will always exist. When we no longer need to support both pipelines I plan on simplifying this workflow a bit, so it will only matter in the interim anyway.